databroker/src/types.rs

```diff
@@ -233,158 +233,21 @@ impl DataValue {
         }
     }
 
-    pub fn less_than(&self, other: &DataValue) -> Result<bool, CastError> {
-        match (&self, other) {
-            (DataValue::Int32(value), DataValue::Int32(other_value)) => Ok(value < other_value),
-            (DataValue::Int32(value), DataValue::Int64(other_value)) => {
-                Ok(i64::from(*value) < *other_value)
-            }
-            (DataValue::Int32(value), DataValue::Uint32(other_value)) => {
-                Ok(i64::from(*value) < i64::from(*other_value))
-            }
-            (DataValue::Int32(value), DataValue::Uint64(other_value)) => {
-                if *value < 0 {
-                    Ok(true) // Negative value must be less than unsigned
-                } else {
-                    match u64::try_from(*value) {
-                        Ok(value) => Ok(value < *other_value),
-                        Err(_) => Err(CastError {}),
-                    }
-                }
-            }
-            (DataValue::Int32(value), DataValue::Float(other_value)) => {
-                Ok(f64::from(*value) < f64::from(*other_value))
-            }
-            (DataValue::Int32(value), DataValue::Double(other_value)) => {
-                Ok(f64::from(*value) < *other_value)
-            }
+    pub fn greater_than_equal(&self, other: &DataValue) -> Result<bool, CastError> {
+        match self.greater_than(other) {
```

Comment:
That is better a question for @argerus. I have some theories - one is that we do not "save" that much memory by using u32 instead of u16/u8 and i32 instead of i8/i16, another is that it simplifies implementation as we do not need to handle that many combinations.

Reply:
https://groups.google.com/g/protobuf-c/c/UzcyFvEOvMY

Reply:
https://protobuf.dev/programming-guides/proto3/#scalar

Reply:
there's no int8 in proto3

Reply:
> That is better a question for @argerus. I have some theories - one is that we do not "save" that much memory by using u32 instead of u16/u8 and i32 instead of i8/i16, another is that it simplifies implementation as we do not need to handle that many combinations.

Exactly. It reduces the combinatorial explosion in the implementation and doesn't really cost anything memory wise since they are all part of the same `enum` anyway. Using 32 & 64 bit integers might even be marginally more efficient cpu wise (probably doesn't matter much). And as @lukasmittag mentions, it would need to be converted anyway when using protobuf.

The VSS data type is enforced when setting a value, and is not encoded by the value enum itself. This simplifies setting a value to `NotAvailable` as well.

Reply:
Got it, thanks